src/utils/workflow-utils.ts

```diff
@@ -380,33 +383,158 @@ export const modifyExistingCaptureDraftSpec = async (
-    constraintType: ConstraintTypes
-): boolean => {
-    const required = isRequireOnlyField(constraintType);
+export const getBindingIndexByResourcePath = <T extends Schema>(
+    resourcePath: string[],
+    schema: Schema
+): number => {
+    if (resourcePath.length === 0) {
+        return -1;
+    }
+
+    const bindings: T[] = schema.bindings;
 
-    return required || constraintType === ConstraintTypes.LOCATION_RECOMMENDED;
+    return bindings.findIndex((binding) => {
+        let bindingResourcePath: string[] = [];
+
+        if (binding?.resourcePath) {
+            bindingResourcePath = binding.resourcePath;
+        } else if (binding?.resource?._meta?.path) {
+            bindingResourcePath = binding.resource._meta.path;
+        }
+
+        return bindingResourcePath.length > 0
+            ? isEqual(bindingResourcePath, resourcePath)
+            : false;
+    });
 };
 
-export const isExcludeOnlyField = (
-    constraintType: ConstraintTypes
-): boolean => {
-    return (
-        constraintType === ConstraintTypes.FIELD_FORBIDDEN ||
-        constraintType === ConstraintTypes.UNSATISFIABLE
-    );
+const getDraftedMaterializationBinding = (
+    draftSpec: Schema,
+    resourcePath: string[] | undefined,
+    stagedBindingIndex: number,
+    targetCollection: string,
+    indexLookupOnly?: boolean
+): {
+    draftedBinding: MaterializationBinding | undefined;
+    draftedBindingIndex: number;
+} => {
+    let draftedBinding: MaterializationBinding | undefined;
+    let draftedBindingIndex = -1;
+
+    // Attempt to identify the target drafted binding by resource path pointer(s) before
+    // relying on the binding index.
+    if (!indexLookupOnly && resourcePath) {
+        draftedBindingIndex =
+            getBindingIndexByResourcePath<MaterializationBinding>(
+                resourcePath,
+                draftSpec
+            );
+
+        draftedBinding = draftSpec.bindings.at(draftedBindingIndex);
+    }
+
+    if (!draftedBinding) {
+        draftedBindingIndex = getBindingIndex(
+            draftSpec.bindings,
+            targetCollection,
+            stagedBindingIndex
+        );
+
+        draftedBinding =
+            draftedBindingIndex > -1
+                ? draftSpec.bindings[draftedBindingIndex]
+                : undefined;
+    }
+
+    return { draftedBinding, draftedBindingIndex };
 };
 
-export const isFieldSelectionType = (value: any): value is FieldSelectionType =>
-    typeof value === 'string' &&
-    (value === 'default' || value === 'exclude' || value === 'require');
+interface BindingTypes<B, V, T> {
+    built: B | undefined;
+    drafted: T | undefined;
+    live: T | undefined;
+    validated: V | undefined;
+}
+
+export interface RelatedBindings {
+    indices: BindingTypes<number, number, number>;
+    values: BindingTypes<
+        BuiltBinding,
+        ValidatedBinding,
+        MaterializationBinding
+    >;
+}
+
+export const getRelatedBindings = (
```

Comment:
I don't see anywhere calling `getRelatedBindings` ... either mark remove to mark with `TODO` on how we'll use in the future.